### ISSUE TITLE
config-distros.sh - Ya gotta watch out for them linuxisms --

### ISSUE THREAD

**Issue by aabatpurdue** [mintty/wsltty#121]
At least two problems with `config-distros.sh`:

(1) `PATH=/bin:$PATH` (twice even (:-}) ).  `/bin` is the (my Ubuntu 18.04) linux base bin directory. Maybe `bin` would be better to get the local `bin` folder/directory?  Or maybe even an absolute `*/appdata/*/bin` path.

(2) Linux will accept an executable with the `[regtool].exe` suffix but it will not even find the process if just specified as `regtool`.  In `config-distros.sh` you could rename the `regtool` function `regtool.exe` and replace all references to `regtool` with `regtool.exe`.  Or simply just rename `regtool.exe` to `regtool`.

(3) Another one.  The shebang says `/bin/sh` but you explicitly mention `dash` within `config-distros.sh`.

**Comment by mintty** [comment]
Thanks; note the script is to be called in a cygwin environment, not under Linux; I don't think it would run under WSL.
(1) $PATH should probably be quoted, changing and removing the duplicate. /bin should indeed be the cygwin bin directory; do you otherwise successfully run the script from WSL?
(2) This also applies to reg.exe, cmd.exe, cscript.exe... To be changed consistently if the script ever gets ported to WSL.
(3) /bin/sh is the generic shell that should always exist, to run the script as a command. From the installation environment, it is explicitly called with dash, not for the sake of the script (which is on the other hand adapted to be compatible with dash) but simply to reduce the installation to dash.exe, not bash.exe with its dependencies.

**Comment by mintty** [comment]
Further (1) Adding /bin to PATH is needed to support direct invocation from install.bat.
If you'd run the script from WSL (assuming that works at all), it's hard to determine the respective cygwin /bin anyway, to be added to PATH, so the assumption would be that it's already in the PATH, like after running wslbridge from an interactive cygwin session. If you run WSL directly from Windows, I see no reliable way to provide access to the cygwin tools needed by the script.

**Comment by aabatpurdue** [comment]
FWIW - I went to the `C:/Users/me/AppData/Local/wsltty` folder/directory and used the WSL shortcut in the right-click context.  My HOME there is my Ubuntu HOME so the `/bin` is the Ubuntu `/bin`.  For me there, I would have to add `/mnt/c/cygwin64/bin` to my PATH to access the Cygwin stuff.

I see the `cygwin1.dll` in the `bin` sub-directory but I would have to add `bin` to my PATH in order to use it.  I did not have any problems until I attempted to mix and match - `ldd(ubuntu) bin/mintty.exe(cygwin?) - `ldd` did not recognize the "not dynamic" pe32+ file type.  Gee, I wonder why (;-}).

As a side note, you still have to add `bin` to the PATH so that `regtool[.exe]` can be used.

Not that I necessarily need to do so but do any of the shortcuts eliminate the need to have the Ubuntu sub-system as a companion?  I was wondering if I could use `mintty` with HOME = my windows HOME even if Ubuntu was removed.

**Comment by mintty** [comment]
To run the config script, you would normally click the "configure WSL shortcuts" or *" context menu" links, and not run it from a WSL session. Both config links and session/terminal links should normally also be installed in your Start folder. The interworking problems you outline do not arise if you use this kind of invocation as it is designed.
With wsltty, a small standalone cygwin installation is included and you would also normally not access a parallel full cygwin installation at the same time, even if you have one.
The wsltty standalone installation is tailored at the WSL target. It is possible to run its mintty.exe with its included dash.exe, for testing purposes. You would not have a really useful cygwin environment then, however. You could even run it with cmd.exe, missing winpty however in that case, so also giving you limited functionality.

**Comment by aabatpurdue** [comment]
This is an FYI.

I have the McAfee _Endpoint Security_ installed (sadly a little old).  I have encountered two differences of opinion between it and WSL products - wsltty included.  It blocked all attempts to use WSL so I disabled that feature.  Yesterday, I found that I could not instantiate `Ubuntu-18.04~.bat`.  Today I found that it did not like the `.04~.bat` suspicious double extension (its words) and was blocking the execution.  I attempted to exclude the file but had no luck doing it.  So I renamed the file `UbuntuT-18.04.bat` (the **T** for twiddle (:-}) ) and I can now instantiate it.  Unfortunately, I suspect that I will run afoul with some of your other code that requires the original name.
